plugin/settings/settings_storage.py

```diff
@@ -0,0 +1,209 @@
+        are not valid.
+
+        Returns:
+            bool: validity of settings
+        """
+        for key, value in self.__dict__.items():
+            if key.startswith('__') or callable(key):
+                continue
+            if value is None:
+                log.critical(" no setting '%s' found!", key)
+                return False
+        if self.cmake_flags_priority not in SettingsStorage.CMAKE_PRIORITIES:
+            log.critical(" priority: '%s' is not one of allowed ones!",
+                         self.cmake_flags_priority)
+            return False
+        return True
+
+    def __load_vars_from_settings(self, settings, project_specific=False):
+        """ Load all settings and add them as attributes of self
+
+        Args:
+            settings (dict): settings from sublime
+            project_specific (bool, optional): Description
+
+        Deleted Parameters:
+            prefixes (list, optional): package-specific prefixes to
+                disambiguate settings when loading them from project settings
+
+        """
+        if project_specific:
+            log.debug(" Overriding settings by project ones if needed:")
+            log.debug(" Valid prefixes: %s", SettingsStorage.PREFIXES)
+        log.debug(" Reading settings...")
+        # project settings are all prefixed to disambiguate them from others
+        if project_specific:
+            prefixes = SettingsStorage.PREFIXES
+        else:
+            prefixes = [""]
+        for setting_name in SettingsStorage.NAMES_ENUM:
+            for prefix in prefixes:
+                val = settings.get(prefix + setting_name)
+                if val is not None:
+                    # we don't want to override existing setting
+                    break
+            if val is not None:
+                # set this value to this object too
+                setattr(self, setting_name, val)
+                # tell the user what we have done
+                log.debug("  %-26s <-- '%s'", setting_name, val)
+        log.debug(" Settings sucessfully read...")
+
+        # initialize max_tu_age if is it not yet, default to 30 minutes
+        self.max_tu_age = getattr(self, "max_tu_age", "00:30:00")
+        # get seconds from string if needed
+        if isinstance(self.max_tu_age, str):
+            self.max_tu_age = Tools.seconds_from_string(self.max_tu_age)
+
+    def __populate_common_flags(self, view):
+        """ Populate the variables inside common_flags with real values
+
+        Args:
+            view (sublime.View): current view
+
+        Returns:
+            str[]: clang common flags with variables expanded
+
+        """
+        # init current and parrent folders:
+        if not view.file_name():
+            log.error(" no view to populate common flags from")
+            return
+        file_current_folder = path.dirname(view.file_name())
+        file_parent_folder = path.dirname(file_current_folder)
```

Comment:
Can you move those two variables down, inside the 'if' conditions, where those are actually used?